src/components/CartItem.js

```diff
@@ -0,0 +1,42 @@
+const CartItem = props => {
+
+    const lowerQuantity = e => {
+        const quantityDecrease = props.cartItems.map(item => item === e ? { ...item, quantity: item.quantity - 1 } : item)
+        props.setCartItems(quantityDecrease)
+
+        for (const items of quantityDecrease) {
+            items.totalPrice = items.basePrice * items.quantity
```

Comment:
Interesting the you calculate a total price here and remove the item if this is 0 - why not just check if quantity === 0?

Reply:
yup i can definitely just check for quantity 0! 
But the calculation of the total price is to update the total price value in the state object itself when clicked on the minus quantity button. but changing to quantity === 0 is probably better for readability